Add Marketplace documentation (#10)
mautic/developer-documentation-new: docs/marketplace/listing.rst

.. code-block:: json

    {
        "name": "example-vendor/plugin-example",
        "description": "Example Plugin",
        "type": "mautic-plugin",
        "version": "1.0",
        "keywords": ["mautic","plugin","integration"],
        "extra": {
            "install-directory-name": "ExampleBundle"
        },
        "require": {
            "php": ">=7.4.0 <8.1",
            "ext-zip": "^1.15",
            "mautic/core-lib": "^4.0"
        }
    }

In addition to the information you already have in the composer.json for your plugin, please add:

- ``type`` must have value ``mautic-plugin``. The Marketplace is filtering PHP packages by this tag. It is required to show up in the Marketplace.
- ``extra.install-directory-name`` specifies the directory name for the bundle. Correct directory name is important for PSR4 autoloading. It must be the same as in the namespace in your PHP classes.
- ``require.php`` be sure to specify the PHP version range that you test the plugin against. Ideally it should be the same as the Mautic's `supported PHP versions <https://www.mautic.org/download/requirements>`_. Do not allow users to install the plugin on versions that are not supported.

> - ``require.ext-*`` if your plugin require some PHP extension, please list them in the require section too. No one knows on what servers your plugin will run on.

Reviewer: ```suggestion
- ``require.ext-*`` if your Plugin require some PHP extension, please list them in the require section too. As servers can have a wide variety of extensions enabled, this ensures that the correct extensions are available.
```

> - ``require.mautic/core-lib`` it's important to specify which Mautic versions your plugin supports. Include only the versions you or your community have tested and confirmed the plugin works with.

Reviewer: ```suggestion
- ``require.mautic/core-lib`` it's important to specify which Mautic versions your Plugin supports. Include only the versions you or your community have tested and confirmed the Plugin works with.
```

> Step 2: check best practices

Reviewer: ```suggestion
Step 2: verify best practices
```

----------------------------

> There's a few things we highly recommend when creating plugins for Mautic. Please refer to the :doc:`./best_practices` page for more details.

Reviewer: ```suggestion
There's a few things which are highly recommend when creating Plugins for Mautic. Please refer to the :doc:`./best_practices` page for more details.
```

Step 3: publish to Composer
---------------------------

> When the composer.json is ready, follow the `Publishing Packages <https://packagist.org>`_ section directly in Packagist.

Reviewer: Please use the external linking extension here!

Step 4: apply for the allowlist
-------------------------------

> While the Mautic Marketplace is still in beta, you'll have to apply for our allowlist in order to show up in the Marketplace. We expect to move to stable with Mautic 4.2 in February 2022, after which the allowlist likely won't be needed anymore. Please refer to the :doc:`./allowlist_what_and_why` page for more details.

Reviewer: ```suggestion
While the Mautic Marketplace is still in beta, you'll have to apply for our allowlist in order to show up in the Marketplace. It is expected that the Marketplace will move from beta to stable release with Mautic 4.2 in February 2022, after which the allowlist may be deprecated in favour of a blocklist. Please refer to the :doc:`./allowlist_what_and_why` page for more details.
```

> Updating existing plugins for usage with the Marketplace

Reviewer: ```suggestion
Updating existing Plugins for usage with the Marketplace
```

========================================================

If you required ``mautic/composer-plugin`` in your plugin's dependencies in the past, please remove it. Support for Mautic plugins is now built into Composer, so you only have to set the type to ``mautic-plugin`` and Composer will automatically install your plugin into the ``plugins`` folder.

> Next to that, if you built your plugin for Mautic 3 originally, please check our `UPGRADE-4.0.md guide <https://github.com/mautic/mautic/blob/4.x/UPGRADE-4.0.md>`_ for the breaking changes we made in Mautic 4.

Reviewer: ```suggestion
Next to that, if you built your plugin for Mautic 3 originally, please check the `UPGRADE-4.0.md guide <https://github.com/mautic/mautic/blob/4.x/UPGRADE-4.0.md>`_ for the breaking changes in Mautic 4.
```
